### ISSUE TITLE
Estimating 2D image gradients with Nerf

### ISSUE THREAD

**Issue by vincentcartillier** [NVlabs/instant-ngp#1272]
Since Nerf is differentiable all the way, can we compute the following Jacobian of an image?
$$J_{\vec{u}} I(\vec{u})$$

where $I: \mathbb{R}^2 \rightarrow \mathbb{R}^3$ would be the Nerf function and $\vec{u} \in \mathbb{R}^2$ a 2D pixel coordinate. 

I am trying to compute this given the current implementation. The hope is to be able to retreive something close to what we would have if we'd apply the Sobel operator on the image directly.

More precisely, I have trained i-NGP on one image. And now I am trying to get the gradient of $I$ w.r.t $\vec{u}$ for all pixels in the image. 

The process is similar to the [train_nerf_step](https://github.com/NVlabs/instant-ngp/blob/aab2886bcb96b725c1f3a0e7f3e411ae4921a1d4/src/testbed_nerf.cu#L3194) function. However, In this case I define the loss function as follow:
$$L = R+G+B $$
with R, G, B the red, green and blue values reconstructed for a given ray. The associated partial derivative w.r.t R, G and B is 1. 

In the end I grab the Jacobian of the loss w.r.t to the pixel locations from the [compute_cam_gradient_train_nerf](https://github.com/NVlabs/instant-ngp/blob/aab2886bcb96b725c1f3a0e7f3e411ae4921a1d4/src/testbed_nerf.cu#L1671) function. More precisely [here](https://github.com/NVlabs/instant-ngp/blob/aab2886bcb96b725c1f3a0e7f3e411ae4921a1d4/src/testbed_nerf.cu#L1750):
```		// Projection of the raydir gradient onto the plane normal to raydir,
		// because that's the only degree of motion that the raydir has.
		vec3 orthogonal_ray_gradient = ray_gradient.d - ray.d * dot(ray_gradient.d, ray.d);

		// Rotate ray gradient to obtain image plane gradient.
		// This has the effect of projecting the (already projected) ray gradient from the
		// tangent plane of the sphere onto the image plane (which is correct!).
		vec3 image_plane_gradient = inverse(mat3(xform)) * orthogonal_ray_gradient;

```

Here is the results I get (left is reconstructed RGB and right is estimated image gradients): 
![image_gradient](https://user-images.githubusercontent.com/42780642/222939807-059b5e31-89cc-4a9b-a10d-03027c2492db.jpg)

Clearly the gradient image seems wrong. 

What am I missing here?
